File: Java/Rectangle.java
public class Rectangle extends ConvexPolygon {
    static final int NUM_SIDES = 4;

    protected double width;
    protected double height;

    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }
    public int getNumSides() {
        return NUM_SIDES;
    }
    public Circle circumscribe() {
        double diagonal = Math.hypot(width, height);
        return new Circle(diagonal/2);
    }
}

File: C++/rectangle.cc
#include "rectangle.h"

static const int NUM_SIDES = 4;

Rectangle::Rectangle(float width, float height)
    : width_(width), height_(height) {
}

int Rectangle::getNumSides() const {
    return NUM_SIDES;
}

float Rectangle::getPerimeter() const {
    return 2 * (width_ + height_);
}